### PR TITLE
Connection information

### PR DESCRIPTION
**Pull request by jelu**
- Add `conn_info` object, UDP/TCP/TLS/QUIC connection information including resumption and latency

### CODE REVIEW

**Comment by jelu** [comment]
@pspacek @nicki-krizek This is my suggestion on how to add connection information into a stats object.

It uses the same way as the other objects that it's based on `"type": "..."` which means there can only be one.

They also inherit each other, so TCP has all the props from UDP, TLS from TCP etc etc.

File: dns-metrics.schema.json [hunk 1]
```diff
@@ -171,9 +171,160 @@
...
+                }
+            }
+        },
+        "tcp_conn": {
+            "$ref": "#/$defs/udp_conn",
+            "title": "DNS Metric TCP Connection Information",
+            "description": "An object with the metrics around the TCP connections used to sent and received for the given stats object",
+            "properties": {
+                "handshakes": {
+                    "description": "The number of handshakes performed",
```
Comment (pspacek) [inline dns-metrics.schema.json:206]: performed = successful? or attempted?

File: dns-metrics.schema.json [hunk 2]
```diff
@@ -171,9 +171,160 @@
...
+                    "type": "array",
+                    "items": {
+                        "type": "array",
+                        "items": {
+                            "type": "integer"
+                        }
+                    }
+                }
+            }
+        },
```
Comment (pspacek) [inline dns-metrics.schema.json:245]: Maybe we can have a generic "latency" object in the schema and reference it from all other places? It should be the same for DNS queries as well, I think.

**Comment by pspacek** [comment]
Thank you! In general it makes sense. I have to run away now but I will have a look at the rendered docs afterwards - probably tomorrow.

File: dns-metrics.schema.json (hunk 1 above)
Comment (jelu) [reply dns-metrics.schema.json:206]: :shrug: this is cut&paste from @nicki-krizek 's comment

File: dns-metrics.schema.json [hunk 3]
```diff
@@ -171,9 +171,160 @@
...
             },
             "required": [ "since", "until" ]
+        },
+        "udp_conn": {
+            "title": "DNS Metric UDP Connection Information",
+            "description": "An object with the metrics around the UDP connections used to sent and received for the given stats object",
+            "type": "object",
+            "properties": {
+                "packets_sent": {
+                    "description": "The number of packets sent",
```
[the remark this answers is not in the document]
Comment (jelu) [reply dns-metrics.schema.json:187]: I donno, suggestion? It also needs to work when inherit into TCP/TLS... etc

[the remark this answers is not in the document]
Comment (pspacek) [reply dns-metrics.schema.json:191]: I don't think so, it should not be shared. There is no information about number of packets for TCP as normally we don't have any visibility into SYN/ACK/retry magic around TCP.

[the remark this answers is not in the document]
Comment (pspacek) [reply dns-metrics.schema.json:191]: Thread migrated to separate issue: https://github.com/DNS-OARC/dns-metrics/issues/10

File: dns-metrics.schema.json (hunk 1 above)
Comment (pspacek) [reply dns-metrics.schema.json:210]: Thread migrated to: https://github.com/DNS-OARC/dns-metrics/issues/11

File: dns-metrics.schema.json (hunk 2 above)
Comment (pspacek) [reply dns-metrics.schema.json:249]: Thread migrated to: https://github.com/DNS-OARC/dns-metrics/issues/12

**Pull request merged**